app/src/components/avatar.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{
+    components::avatar_event_bus::AvatarEventBus,
+    infra::common_component::{CommonComponent, CommonComponentParts}};
+use anyhow::{bail, Result};
+use graphql_client::GraphQLQuery;
+use yew::{prelude::*};
+use yew_agent::{Bridge, Bridged};
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "queries/get_user_avatar.graphql",
+    response_derives = "Debug",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+pub struct GetUserAvatar;
+
+pub struct Avatar {
+    common: CommonComponentParts<Self>,
+    /// The user info. If none, the error is in `error`. If `error` is None, then we haven't
+    /// received the server response yet.
+    avatar: Option<String>,    
```

Comment:
Can we wrap that in a new type? `Option<AvatarData>` with `pub struct AvatarData(String);`